webodf/lib/ops/OdtDocument.js

```diff
@@ -126,12 +126,16 @@ ops.OdtDocument = function OdtDocument(odfCanvas) {
      * @param {!Element} documentElement
      */
     this.setDocumentElement = function (documentElement) {
-        var odfContainer = odfCanvas.odfContainer();
+        var odfContainer = odfCanvas.odfContainer(),
+            rootNode;
+
         // TODO Replace with a neater hack for reloading the Odt tree
-        // Once this is fixed, SelectionView.addOverlays & OdtStepsTranslator.verifyRootNode can be largely removed
+        // Once this is fixed, SelectionView.addOverlays can be removed
         odfContainer.setRootElement(documentElement);
         odfCanvas.setOdfContainer(odfContainer, true);
         odfCanvas.refreshCSS();
+        rootNode = getRootNode();
+        stepsTranslator = new ops.OdtStepsTranslator(rootNode, createPositionIterator(rootNode), filter, 500);
```

Comment:
Replacing the stepsTranslator instance also needs to care for unsubscribing the old from the steps changes signal and instead subscribing the new to them, cmp. `init()` method.